src/main/java/it/gov/pagopa/payhub/activities/activity/utility/SendEmailIngestionFlowActivityImpl.java

```diff
@@ -0,0 +1,125 @@
+import java.util.HashMap;
+import java.util.Map;
+
+import static it.gov.pagopa.payhub.activities.activity.utility.util.Constants.MAIL_DATE_FORMAT;
+import static it.gov.pagopa.payhub.activities.activity.utility.util.Constants.WS_USER;
+
+/**
+ * Implementation of SendEmailIngestionFlowActivity for send email ingestion flow activity.
+ * Sends an email based on the status of a processed file identified by its IngestionFlow ID.
+ */
+@Lazy
+@Slf4j
+@Component
+public class SendEmailIngestionFlowActivityImpl implements SendEmailIngestionFlowActivity {
+    private final EmailTemplatesConfiguration emailTemplatesConfiguration;
+    private final UserAuthorizationService userAuthorizationService;
+    private final SendMailService sendMailService;
+    private final IngestionFlowFileDao ingestionFlowFileDao;
+
+    @Value("${activity.root.path}")
+    private String fsRootPath;
+
+    public SendEmailIngestionFlowActivityImpl(
+            EmailTemplatesConfiguration emailTemplatesConfiguration,
+            UserAuthorizationService userAuthorizationService,
+            IngestionFlowFileDao ingestionFlowFileDao,
+            SendMailService sendMailService) {
+        this.emailTemplatesConfiguration = emailTemplatesConfiguration;
+        this.userAuthorizationService = userAuthorizationService;
+        this.ingestionFlowFileDao  = ingestionFlowFileDao;
+        this.sendMailService = sendMailService;
+    }
+
+    /**
+     * Sends an email based on the process result of the given file ingestionFlow ID.
+     *
+     * @param ingestionFlowFileId       the unique identifier of the IngestionFlow record related to the imported file.
+     * @param success      true if the process succeeded, false otherwise.
+     * @return true if the email was sent successfully, false otherwise.
+     */
+    @Override
+    public boolean sendEmail(Long ingestionFlowFileId, boolean success) {
+
+        try {
+            IngestionFlowFileDTO ingestionFlowFileDTO = ingestionFlowFileDao.findById(ingestionFlowFileId)
+                    .orElseThrow(() -> new IngestionFlowNotFoundException("Cannot found ingestionFlow having id: "+ ingestionFlowFileId));
+            String ipaCode = ingestionFlowFileDTO.getOrg().getIpaCode();
+            UserInfoDTO userInfoDTO = userAuthorizationService.getUserInfo(ipaCode, ingestionFlowFileDTO.getOperatorName());
+            UserInfoDTO organizationUserInfo = userAuthorizationService.getUserInfo(ipaCode,ipaCode+WS_USER);
+            MailTo mailTo = getMailFromIngestionFlow(ingestionFlowFileDTO, success);
+            mailTo.setTo(new String[]{userInfoDTO.getEmail()});
+            if (organizationUserInfo!= null && StringUtils.isNotBlank(organizationUserInfo.getEmail()) &&
+                ! organizationUserInfo.getEmail().equalsIgnoreCase(userInfoDTO.getEmail())) {
+                    mailTo.setCc(new String[]{organizationUserInfo.getEmail()});
+            }
+            mailTo.setHtmlText(mailTo.getHtmlText());
```

Comment:
what is the purpose of this line? :)

Reply:
removed useless line of code (inserted during refactoring)